modules/xphoto/src/bm3d_image_denoising.cpp

```diff
@@ -148,6 +148,7 @@ void bm3dDenoising(
         _dst.create(srcSize, type);
         break;
     case BM3D_STEPALL:
+        _basic.create(srcSize, type);
         _dst.create(srcSize, type);
```

Comment:
Please take a look on line 159.

Perhaps this should be guarded: `if (_basic.needed()) basic.create(...);`